### PR TITLE
Update parameter names

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes issue #451 

Summary of changes in this pull request:

* energy -> flow
* resource -> source/sink
* force_resource -> _min/_max/_equals on source/sink
* demands (via `sinks`) & `flow_in` are now positive

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

**Comment by brynpickering** [comment]
I expect some tests to fail as I originally built this off #492, so some tests expect logging outputs that look different. 

Future PRs will build off the combination of this PR and #492.

File: doc/_static/notebooks/calliope_model_object.ipynb
[the remark this answers is not in the document]
Comment (brynpickering) [reply doc/_static/notebooks/calliope_model_object.ipynb:1]: see #495

**Comment by brynpickering** [comment]
I don't get the docs failure locally, so I'm hoping it's a random glitch

**Review by sjpfenninger — APPROVED** [review]

**Comment by brynpickering** [comment]
Error is caused by mamba ignoring `ruamel.yaml < 0.18` and seeing it instead as `ruamel.yaml <= 0.18`. Pinned for extra security.

**Review by sjpfenninger — APPROVED** [review]

**Pull request merged**